zerver/tests/test_message_edit.py

```diff
@@ -1596,3 +1598,100 @@ def test_user_group_mention_restrictions_while_editing(self) -> None:
             },
         )
         self.assert_json_success(result)
+
+    def test_move_message_to_new_topic_with_automatic_follow_policy(self) -> None:
+        self.login("iago")
+        hamlet = self.example_user("hamlet")
+        shiva = self.example_user("shiva")
+        stream = self.make_stream("new_stream")
+        self.subscribe(hamlet, stream.name)
+        self.subscribe(shiva, stream.name)
+
+        self.send_stream_message(
+            sender=hamlet,
+            stream_name="new_stream",
+            topic_name="test",
+            content="Never here",
+        )
+        msg_id = self.send_stream_message(
+            sender=hamlet,
+            stream_name="new_stream",
+            topic_name="test",
+            content="Always here",
+        )
+
+        do_change_user_setting(
+            hamlet,
+            "automatically_follow_topics_policy",
+            UserProfile.AUTOMATICALLY_CHANGE_VISIBILITY_POLICY_ON_INITIATION,
+            acting_user=None,
+        )
+        result = self.client_patch(
+            f"/json/messages/{msg_id}",
+            {
+                "topic": "testing",
+                "propagate_mode": "change_one",
+                "send_notification_to_old_thread": "false",
+            },
+        )
+        self.assert_json_success(result)
+
+        stream_topic_target = StreamTopicTarget(
+            stream_id=stream.id,
+            topic_name="testing",
+        )
+        user_ids = stream_topic_target.user_ids_with_visibility_policy(
```

Comment:
I think the right pattern to test here would be:

* assert that user was not following the topic
* do the move topic operation
* assert that user now follows.

The reason to do that first step is to avoid a case where the topic is already followed (due to some reason) -- then the test after that would be of no use for us. :)

Reply:
updated